SCons/Util/stats.py

```diff
@@ -0,0 +1,212 @@
+        for k in sorted(self.stats_table.keys()):
+            r = self.stats_table[k][:l] + [k]
+            self.outfp.write(fmt2 % tuple(r))
+
+
+class MemStats(Stats):
+    def do_append(self, label):
+        self.labels.append(label)
+        self.stats.append(SCons.Debug.memory())
+
+    def do_print(self):
+        fmt = 'Memory %-32s %12d\n'
+        for label, stats in zip(self.labels, self.stats):
+            self.outfp.write(fmt % (label, stats))
+
+
+class TimeStats(Stats):
+    def __init__(self):
+        super().__init__()
+        self.totals = {}
+        self.commands = {}  # we get order from insertion order, and can address individual via dict
+
+    def total_times(self, build_time, sconscript_time, scons_exec_time, command_exec_time):
+        self.totals = {
+            'build_time': build_time,
+            'sconscript_time': sconscript_time,
+            'scons_exec_time': scons_exec_time,
+            'command_exec_time': command_exec_time
+        }
+
+    def add_command(self, command, start_time, finish_time):
+        if command in self.commands:
+            print("Duplicate command %s" % command)
+        self.commands[command] = {'start': start_time,
+                                  'end' : finish_time,
+                                  'duration': finish_time - start_time}
+
+
+COUNT_STATS = CountStats()
+MEMORY_STATS = MemStats()
+TIME_STATS = TimeStats()
+
+
+
+def WriteJsonFile():
+    """
+    Actually write the JSON file with debug information.
+    Depending which of : count, time, action-timestamps,memory their information will be written.
+    """
+
+    from SCons.Script import BUILD_TARGETS, COMMAND_LINE_TARGETS, ARGUMENTS, ARGLIST
+
+    # print(f"DUMPING JSON FILE: {JSON_OUTPUT_FILE}")
+    json_structure = {}
+    if COUNT_STATS.enabled:
+        json_structure['Object counts'] = {}
+
+        oc = json_structure['Object counts']
+        for c in COUNT_STATS.stats_table:
+            oc[c] = {}
+            for l, v in zip(COUNT_STATS.labels, COUNT_STATS.stats_table[c]):
+                oc[c][''.join(l)] = v
+
+    if MEMORY_STATS.enabled:
+        json_structure['Memory'] = {}
+
+        m = json_structure['Memory']
+        for label, stats in zip(MEMORY_STATS.labels, MEMORY_STATS.stats):
+            m[label] = stats
+
+    if TIME_STATS.enabled:
+        json_structure['Time'] = {'Commands': TIME_STATS.commands,
+                                  'Totals': TIME_STATS.totals}
+
+    # Now add information about this build to the JSON file
+    json_structure['Build_Info'] = {
+        'BUILD_TARGETS' : [str(t) for t in BUILD_TARGETS],
+        'ARGUMENTS' : [str(a) for a in ARGUMENTS],
+        'ARGLIST' : [ str(al) for al in ARGLIST],
+        'COMMAND_LINE_TARGETS' : [ str(clt) for clt in COMMAND_LINE_TARGETS],
+        'ARGV' : sys.argv,
+        'TIME' : datetime.now().isoformat(),
+        'HOST' : platform.node(),
+        'PYTHON_VERSION' : {
```

Comment:
I'd use `sysconfig` to pull the info (more "standardized" that way) and include more details, although I don't specifically know which ones to add. Would people use this to compare behavior of different Python builds?

Reply:
potentially you're storing these for a farm of build machines.. this way you could identify one being slower/faster and investigate.. at least that's what I was thinking.
Seemed simple enough to add it now, and could be useful..

Reply:
sure, I was just suggesting more info might be good - or maybe add it later. For fun, do `python -m sysconfig` ... except on Windows, there's a TON of info captured now.

Reply:
Ugh.. that's too much info... I think I'll leave it as is for now.. ;)